### PR TITLE
Ensure rails 7.2 compatibility

### LINKED ISSUE #1685: `require 'factory_bot'` fails with latest ActiveSupport gem (7.2.0)
Per https://github.com/rails/rails/issues/52582 one must `require "active_support"` before requiring any other definition(s) like `active_support/core_ext/module/delegation`.

Seems like the fix would be to add `require "active_supoet"` just before line 2 of [lib/factory_bot.rb](https://github.com/thoughtbot/factory_bot/blob/main/lib/factory_bot.rb#L2-L5)

### Description

```
~/.gem/ruby/3.2.5/gems/activesupport-7.2.0/lib/active_support/delegation.rb:47:in `generate': uninitialized constant #<Class:ActiveSupport::Delegation>::Inflector (NameError)
```

### Reproduction Steps

Create `foo.rb` with the following contents:
```ruby
require 'factory_bot'

puts 'Success!'
```

Then invoke the file with ruby

### Expected behavior

Should behave the same as with earlier versions of ActiveSupport:

```shell
; gem list activesupport

*** LOCAL GEMS ***

activesupport (7.1.3.4)

; ruby foo.rb
Success!
```

### Actual behavior

```shell
; gem list activesupport

*** LOCAL GEMS ***

activesupport (7.2.0)

; ruby foo.rb
~/.gem/ruby/3.2.5/gems/activesupport-7.2.0/lib/active_support/delegation.rb:47:in `generate': uninitialized constant #<Class:ActiveSupport::Delegation>::Inflector (NameError)

          unless Inflector.safe_constantize(to.name).equal?(to)
                 ^^^^^^^^^
	from ~/.gem/ruby/3.2.5/gems/activesupport-7.2.0/lib/active_support/core_ext/module/delegation.rb:161:in `delegate'
	from ~/.gem/ruby/3.2.5/gems/factory_bot-6.4.6/lib/factory_bot/definition_hierarchy.rb:3:in `<class:DefinitionHierarchy>'
	from ~/.gem/ruby/3.2.5/gems/factory_bot-6.4.6/lib/factory_bot/definition_hierarchy.rb:2:in `<module:FactoryBot>'
	from ~/.gem/ruby/3.2.5/gems/factory_bot-6.4.6/lib/factory_bot/definition_hierarchy.rb:1:in `<top (required)>'
	from <internal:~/.rubies/ruby-3.2.5/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:97:in `require'
	from <internal:~/.rubies/ruby-3.2.5/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:97:in `require'
	from ~/.gem/ruby/3.2.5/gems/factory_bot-6.4.6/lib/factory_bot.rb:8:in `<top (required)>'
	from <internal:~/.rubies/ruby-3.2.5/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:160:in `require'
	from <internal:~/.rubies/ruby-3.2.5/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:160:in `rescue in require'
	from <internal:~/.rubies/ruby-3.2.5/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:40:in `require'
	from foo.rb:1:in `<main>'
<internal:~/.rubies/ruby-3.2.5/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require': cannot load such file -- factory_bot (LoadError)
	from <internal:~/.rubies/ruby-3.2.5/lib/ruby/3.2.0/rubygems/core_ext/kernel_require.rb>:86:in `require'
	from foo.rb:1:in `<main>'
```

### System configuration
**factory_bot version**:  `factory_bot (6.4.6)`
**rails version**:  n/a  `activesupport (7.2.0)`
**ruby version**: `ruby-3.2.5`

### PR DESCRIPTION
**Pull request by Earlopain**
Closes #1685

This whole thing basically boils down to adding `require "active_support"`.

Additionally, I did the following:
* Bumped development dependencies. Older versions of `sqlite3` fail to compile with gcc 14
* Locked `sqlite3` to `~> 1.4`. Rails 7.1 and lower don't accept 2.0
* Added rails 7.2 to CI
* Resolved a deprecation warning in tests surrounding enums for rails 7.2 and main

### CODE REVIEW

**Comment by rebase-master** [comment]
@sarahraqueld @smaboshe could you please review and merge if the changes look good?

File: spec/acceptance/enum_traits_spec.rb [hunk 1]
```diff
@@ -1,9 +1,17 @@
 describe "enum traits" do
+  def define_model_with_enum(class_name, field, values)
+    define_model(class_name, status: :integer) do
+      if ActiveRecord::VERSION::STRING >= "7.2"
```
Comment (neilvcarvalho) [inline spec/acceptance/enum_traits_spec.rb:4]:
```suggestion
      if ActiveRecord::VERSION::STRING >= "7.0"
```

While the old syntax was removed on Rails 7.2, the new one was introduced on 7.0. By referencing the version when it was added we can more easily remove the old syntax when we drop support for Rails < 7.0.

(Rails 6.1 will stop receiving security updates in a month or so, by the way)

**Review by neilvcarvalho — APPROVED** [review]
(not a maintainer)

Looks great! I just added a minor suggestion for the version evaluation.

File: spec/acceptance/enum_traits_spec.rb (hunk 1 above)
Comment (neilvcarvalho) [reply spec/acceptance/enum_traits_spec.rb:4]: Oh, it was just deprecated, not removed. I thought it was deprecated when the new syntax landed.

Comment (Earlopain) [reply spec/acceptance/enum_traits_spec.rb]: Makes sense, thanks. I applied your suggestion.

**Review by smaboshe — APPROVED** [review]
Thank you so much!

**Pull request merged**